File: Printpress.Application/SpareParts/DTO/SparePartPurchaseInvoiceCreateDto.cs
namespace Printpress.Application;

public class SparePartPurchaseInvoiceCreateDto
{
    public string InvoiceNumber { get; set; }
    public DateTime InvoiceDate { get; set; }
    public string SupplierName { get; set; }
    public string AttachmentFilePath { get; set; }
    public List<SparePartPurchaseInvoiceLineCreateDto> Lines { get; set; }
}

public class SparePartPurchaseInvoiceLineCreateDto
{
    public Guid SparePartItemId { get; set; }
    public decimal Quantity { get; set; }
    public decimal UnitPrice { get; set; }
}

File: Printpress.UI/Printpress_Angular/src/app/features/spare-parts/models/spare-part-purchase-invoice-create.dto.ts
export interface SparePartPurchaseInvoiceLineCreateDto {
  sparePartItemId: string;
  quantity: number;
  unitPrice: number;
}

export interface SparePartPurchaseInvoiceCreateDto {
  invoiceDate: string;
  supplierName: string;
  attachmentFilePath: string;
  paidNow: number;
  receiveNow: boolean;
  lines: SparePartPurchaseInvoiceLineCreateDto[];
}
